Menu and tagging

pbywater opened the pull request:
- Added shuffle and search buttons
- Animation on tag/search menu
- Added unique tags based on data
- Added ability to filter by tag by clicking on an individual tag or by removing tags
- Added ability to clear filter
#14

joeylouise committed:
Add html and css for sticky menu
Relates #15

joeylouise committed:
Add functions to open individual items within menu
Relates #15

joeylouise committed:
Merge branch 'master' into style-menu

pbywater committed:
adding tagging functionality
Integrating old code we wrote for showing nodes by tag

relates to #7

pbywater committed:
moving tag sorting logic into helper function

relates to #7

joeylouise committed:
Add ds store and assets to gitignore #15

joeylouise committed:
Merging style-menu with adding-tagging
relates #15

joeylouise committed:
Remove link to margins.js as this file no longer exists
Relates #15

joeylouise committed:
Add icons folder to assets
Relates #15

joeylouise committed:
Add all icons to menu
Rename take photo icons folders
Relates #15

pbywater committed:
adding contents of top bar

relates to #7 #14

pbywater committed:
adding top buttons

relates to #14

pbywater committed:
getting rid of excess svg code

relates to #14

pbywater committed:
rotates on click

relates to #14

pbywater committed:
tags now showing

relates to 14

pbywater committed:
adding cdn for quicksand font

relates to #14

pbywater committed:
switched in closing icon and tags now display

relates to #14

pbywater committed:
tagging animations finished and working

relates to #14

pbywater committed:
resolving conflicts

relates to #14

joeylouise committed:
Remove references to hamburger class in main.js
Relates #15

joeylouise committed:
Add svg elements instead of img tags as we need to be able to change the colour of the svg in an animation
Add close_icon for meun
Relates #15

pbywater committed:
fixing bug on closing tag nav. Nodes now dont disappear

relates to #14

joeylouise committed:
Change menu modals into rectangles with 3px border-radius instead of ovals
Add colour change to menu icon svgs during transition
Relates #15

pbywater committed:
fixing tag error

relates to #14

joeylouise committed:
Remove references to hamburger classes
Add colour change to menu icon svg
Change translate3d to translate
Relates #15

pbywater committed:
tags are now hidden when inactive

relates to #14

joeylouise committed:
Change transform translate for all menu icon animations at 100%
Relates #15

joeylouise committed:
Fix merge conflicts in src/client/helpers/helpers.js
Relates #15

joeylouise committed:
Remove comments in code
Relates #15

pbywater committed:
tags now can be removed

relates to #14

alice-carr committed:
cycled through source node ids
relates #52

alice-carr committed:
took in source nodes and linked up by id
relates #52

pbywater committed:
tidying up functions

relates to #14

pbywater committed:
merging

relates to #14

pbywater committed:
fixing html ordering

relates to #14

pbywater committed:
removing comment in css

relates to #14

joeylouise commented:
This PR also
-adds the sticky menu to the homepage
-adds all icons into assets folder

Relates #15 #7

alice-carr committed:
merging with master

alice-carr committed:
Merge branch 'master' into menu-and-tagging

ghost commented:
Also links up the main three 'source' nodes to one another
relates #52

des-des reviewed (changes requested):
Remove all the DS_Store files, change your git ignore so these do not come up again..
`public/assets/icons/menu/notification/_DS_Store`

des-des commented on .gitignore:
```diff
@@ -59,6 +59,11 @@ typings/
 config.env
 
+# DS store
+.DS_Store
```
**/*DS_Store

might do it ..?

pbywater committed:
changing opacity on nodes

des-des commented on public/main.css:
```diff
@@ -34,6 +38,393 @@
+  text-align: center;
+  width: 100%;
+  bottom: 0;
+  z-index: 999999;
```
`z-index: 2` is prob okay

pbywater replied:
Haha I'll change these

des-des commented on public/main.css:
```diff
@@ -34,6 +38,393 @@
+ .tag-container {
+   text-align: center;
+   height: 50%;
+     display: none;
```
indentaion

des-des commented on public/main.css:
```diff
@@ -34,6 +38,393 @@
+
+
+ .tags li {
+      list-style-type: none;
```
indentation

des-des commented on public/main.css:
```diff
@@ -34,6 +38,393 @@
+
+.top-bar {
+  position: absolute;
+  z-index: 99999;
```
too big

des-des commented on public/main.css:
```diff
@@ -34,6 +38,393 @@
+.underline {
+  width:90%;
+  height: 90%;
+    display: inline-block;
```
indentation

des-des commented on public/main.css:
```diff
@@ -34,6 +38,393 @@
+
+}
+
+.underline:after {
```
is underline the best class name?

pbywater replied:
It refers to a line that extends underneath the tags - would there be a better name for it?

des-des commented on public/main.css:
```diff
@@ -34,6 +38,393 @@
+}
+
+.filter-tags {
+width: 10%;
```
indentaion

des-des commented on public/main.css:
```diff
@@ -34,6 +38,393 @@
+
+.close-tags {
+  text-align: right;
+    background: transparent !important;
```
indentation

des-des commented on public/scripts/main.js:
```diff
@@ -0,0 +1,84 @@
+document.addEventListener('DOMContentLoaded', (event) => {
```
This file needs to be refactored, heavily. Talk to me if you do not know how

joeylouise replied:
Nice and refactored now! 👍

des-des commented on src/client/d3/d3.js:
```diff
@@ -15,6 +16,20 @@ d3.json(url, (err, data) => {
+  // Add unique tags to tag list for user to select from
+  Object.keys(taggedNodesByTag).forEach((tag) => {
+    $('.tags').append(
+      `<li class='tag-container ${tag}'>
```
Vulnerable to script injection attacks

des-des commented on src/client/helpers/helpers.js:
```diff
@@ -10,6 +10,100 @@ const binByKey = (key, xs) =>
+  $('.close-tags').on('click', () => {
+    $('.underline').removeClass('active');
+    setTimeout(() => {
+      $('.shuffle-memories').fadeIn(1000);
```
You have variables like, 1000, and 500 scattered around everywhere.

Make a variable?

pbywater committed:
Merge branch 'menu-and-tagging' of https://github.com/pbywater/echo into menu-and-tagging

pbywater committed:
changing gitignore for ds store

relates to #14

pbywater committed:
changing z-index to 2

relates to #14

pbywater committed:
fixing indentation

relates to #14

pbywater committed:
adding normal time delay variable

relates to #14

pbywater committed:
regex on tag to avoid script injection

relates to #14

joeylouise committed:
Refactor menu functions in main.js
Remove duplicate tape dependency
Relates #15

joeylouise committed:
Remove unnecessary selectors from main.js
Add if statement to prevent menu button fading out when expanded menu item is clicked
Relates #15

joeylouise committed:
Add watchify and browserify scripts to package.json
Relates #58
